lib/nnc/mfa/ccv_nnc_mfa_gemm.cpp

```diff
@@ -109,7 +109,19 @@ void ccv_nnc_mfa_encode_gemm(mfa::context* context, ccv_nnc_mfa_gemm_params_t pa
         i * byte_stride_d,
       };
     }
-    encoder->setBytes(matrix_offsets, batch_size * 32, 10);
+    if (batch_size * 32 > 4096) {
+#ifdef __x86_64__
+          auto buffer = context->device->newBuffer(batch_size * 32, MTL::ResourceStorageModePrivate);
```

Comment:
newBuffer(matrix_offsets, batch_size * 32, MTL:: ResourceStorageModePrivate); will raise error 
"-[MTLDebugDevice newBufferWithBytes:length:options:]:670: failed assertion `Buffer Validation
storageModePrivate incompatible with ...WithBytes variant of newBuffer"

 auto buffer = context->device->newBuffer(batch_size * 32, MTL::ResourceStorageModePrivate); will raise CPU compatible error
 
current one with manual copy runs without error and eyeball looks good.

Reply:
In that case, I prefer previous code (StorageModeShared).